src/monitord/monitor_actions.c

```diff
@@ -51,7 +51,7 @@ void monitor_agents_disconnection(){
     char str_agent_id[12];
 
     agents_array = wdb_disconnect_agents(time(0) - mond.global.agents_disconnection_time, &sock);
-    if (agents_array) {
+    if (mond.monitor_agents != 0 && agents_array) {
```

Comment:
This will lead to memory leaks when mond.monitor_agents == 0.
I suggest to move the control of mond.monitor_agents outside this method directly

Reply:
Nice! Thank you.